source/exe/process_wide.cc

```diff
@@ -12,10 +12,10 @@ namespace {
 // Static variable to count initialization pairs. For tests like
 // main_common_test, we need to count to avoid double initialization or
 // shutdown.
-uint32_t process_wide_initialized;
+std::atomic<uint32_t> process_wide_initialized;
```

Comment:
strictly speaking std::atomic is a non-POD and we shouldn't have a static one of those.

WDYT about using MUTABLE_CONSTRUCT_ON_FIRST_USE(std::atomic<uint32_t>) for this?

Too hairy?

Too hairy?

Reply:
I'm okay with that, let me push a change